src/nft/hooks/usePermit2Approval.ts

```diff
@@ -3,12 +3,35 @@ import { InterfaceEventName } from '@uniswap/analytics-events'
 import { CurrencyAmount, Token } from '@uniswap/sdk-core'
 import { UNIVERSAL_ROUTER_ADDRESS } from '@uniswap/universal-router-sdk'
 import { useWeb3React } from '@web3-react/core'
+import { usePwatNewContractEnabled } from 'featureFlags/flags/pwatNewContract'
 import usePermit2Allowance, { AllowanceState } from 'hooks/usePermit2Allowance'
 import { useCallback, useMemo, useState } from 'react'
 import invariant from 'tiny-invariant'
 
-// TODO: This should be removed when the sdk is updated to include the new UR address
-const NFT_UNIVERSAL_ROUTER_MAINNET_ADDRESS = '0x4c60051384bd2d3c01bfc845cf5f4b44bcbe9de5'
+const DEPRECATED_NFT_UNIVERSAL_ROUTER_MAINNET_ADDRESS = '0x4c60051384bd2d3c01bfc845cf5f4b44bcbe9de5'
+const NFT_UNIVERSAL_ROUTER_MAINNET_ADDRESS = '0x3fC91A3afd70395Cd496C647d5a6CC9D4B2b7FAD'
+
+const useUniversalRouterAddress = (
+  enabled?: boolean,
+  shouldUseNftRouter?: boolean,
+  chainId?: number
+): string | undefined => {
+  let contractToApprove: string | undefined
+  const shouldUseUpdatedContract = usePwatNewContractEnabled()
+
+  const canPermit = enabled && chainId
+  const forceNftRouter = canPermit && shouldUseNftRouter && chainId === 1
+
+  if (forceNftRouter) {
+    contractToApprove = shouldUseUpdatedContract
+      ? NFT_UNIVERSAL_ROUTER_MAINNET_ADDRESS
+      : DEPRECATED_NFT_UNIVERSAL_ROUTER_MAINNET_ADDRESS
+  } else if (canPermit) {
+    contractToApprove = UNIVERSAL_ROUTER_ADDRESS(chainId)
+  }
+
+  return useMemo(() => contractToApprove, [contractToApprove])
```

Comment:
This is a string, so it does not need to be memo-ized. You can simplify by just returning earli in L25-31.